akka-distributed-data/src/main/scala/akka/cluster/ddata/ORMap.scala

```diff
@@ -228,9 +228,10 @@ final class ORMap[A, B <: ReplicatedData] private[akka] (
           "value, because important history can be lost when replacing the `ORSet` and " +
           "undesired effects of merging will occur. Use `ORMultiMap` or `ORMap.updated` instead.")
     else {
-      val putDeltaOp = PutDeltaOp(keys.resetDelta.add(node, key).delta.get, key → value, zeroTag)
-      // put forcibly damages history, so we propagate full value that will overwrite previous values
-      new ORMap(keys.add(node, key), values.updated(key, value), zeroTag, Some(newDelta(putDeltaOp)))
+      val newKeys = keys.resetDelta.add(node, key)
+      val putDeltaOp = PutDeltaOp(newKeys.delta.get, key → value, zeroTag)
+      // put forcibly damages history, so we consciously propagate full value that will overwrite previous value
+      new ORMap(newKeys, values.updated(key, value), zeroTag, Some(newDelta(putDeltaOp)))
```

Comment:
Change `newKeys` here to `keys.resetDelta.add(node, key)` to see added test fail with
```
Some(55) did not equal Some(56)
```